pkg/sdkserver/sdkserver_test.go

```diff
@@ -1084,6 +1086,377 @@ func TestSDKServerReserveTimeout(t *testing.T) {
+					Name:  "widgets",
+					Count: wrapperspb.Int64(101),
+				}}},
+			want:       agonesv1.CounterStatus{Count: int64(10), Capacity: int64(100)},
+			updateErrs: []bool{true},
+			updated:    false,
+		},
+		"increment past set capacity illegal": {
+			counterName: "widgets",
+			requests: []*alpha.UpdateCounterRequest{{
+				CounterUpdateRequest: &alpha.CounterUpdateRequest{
+					Name:     "widgets",
+					Capacity: wrapperspb.Int64(0),
+				}},
+				{CounterUpdateRequest: &alpha.CounterUpdateRequest{
+					Name:      "widgets",
+					CountDiff: 1,
+				}}},
+			want:       agonesv1.CounterStatus{Count: int64(10), Capacity: int64(0)},
+			updateErrs: []bool{false, true},
+			updated:    true,
+		},
+		"decrement past set capacity OK": {
+			counterName: "widgets",
+			requests: []*alpha.UpdateCounterRequest{{
+				CounterUpdateRequest: &alpha.CounterUpdateRequest{
+					Name:     "widgets",
+					Capacity: wrapperspb.Int64(0),
+				}},
+				{CounterUpdateRequest: &alpha.CounterUpdateRequest{
+					Name:      "widgets",
+					CountDiff: -1,
+				}}},
+			want:       agonesv1.CounterStatus{Count: int64(9), Capacity: int64(0)},
+			updateErrs: []bool{false, false},
+			updated:    true,
+		},
+	}
+
+	for test, testCase := range fixtures {
+		t.Run(test, func(t *testing.T) {
+			m := agtesting.NewMocks()
+
+			m.AgonesClient.AddReactor("list", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				gs := agonesv1.GameServer{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test", Namespace: "default", Generation: 1,
+					},
+					Spec: agonesv1.GameServerSpec{
+						SdkServer: agonesv1.SdkServer{
+							LogLevel: "Debug",
+						},
+					},
+					Status: agonesv1.GameServerStatus{
+						Counters: map[string]agonesv1.CounterStatus{
+							"widgets": {Count: int64(10), Capacity: int64(100)},
+						},
+					},
+				}
+				gs.ApplyDefaults()
+				return true, &agonesv1.GameServerList{Items: []agonesv1.GameServer{gs}}, nil
+			})
+
+			updated := make(chan map[string]agonesv1.CounterStatus, 10)
+			m.AgonesClient.AddReactor("update", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				ua := action.(k8stesting.UpdateAction)
+				gs := ua.GetObject().(*agonesv1.GameServer)
+				gs.ObjectMeta.Generation++
+				updated <- gs.Status.Counters
+				return true, gs, nil
+			})
+
+			ctx, cancel := context.WithCancel(context.Background())
+			sc, err := defaultSidecar(m)
+			require.NoError(t, err)
+			assert.NoError(t, sc.WaitForConnection(ctx))
+			sc.informerFactory.Start(ctx.Done())
+			assert.True(t, cache.WaitForCacheSync(ctx.Done(), sc.gameServerSynced))
+
+			wg := sync.WaitGroup{}
+			wg.Add(1)
+
+			go func() {
+				err = sc.Run(ctx)
+				assert.NoError(t, err)
+				wg.Done()
+			}()
+
+			// check initial value comes through
+			// async, so check after a period
+			err = wait.Poll(time.Second, 10*time.Second, func() (bool, error) {
```

Comment:
Any suggestions for speeding up this test? wait.Poll is slowing the test down.

Reply:
could instead use `require.Eventually()` (which we've been slowly moving over to), which I think won't wait an initial period either.